pkgs/by-name/gt/gtklock-userinfo-module/package.nix

```diff
@@ -0,0 +1,46 @@
+{
+  lib,
+  stdenv,
+  fetchFromGitHub,
+  meson,
+  ninja,
+  pkg-config,
+  gtk3,
+  glib,
+  accountsservice,
+  gtklock,
+}:
+
+stdenv.mkDerivation (finalAttrs: {
+  pname = "gtklock-userinfo-module";
+  version = "4.0.0";
+
+  src = fetchFromGitHub {
+    owner = "jovanlanik";
+    repo = "gtklock-userinfo-module";
+    rev = "v${finalAttrs.version}";
+    hash = "sha256-d9S0Tj7aII2JQ5/PZmt8HaUIb5caoD4GND0PGvuRMn8=";
+  };
+
+  nativeBuildInputs = [
+    meson
+    ninja
+    pkg-config
+  ];
+
+  buildInputs = [
+    gtk3
```

Comment:
Maybe we can have this inside of `gtklock.passthru.module-deps` (or something like that) as it is shared between all modules. I guess `propagatedBuildInputs` would also be an option but I *think* that would also apply to `buildEnv`?

Reply:
I don't like this approach very much, but of course there are different opinions.

For example, `kdePackages` adopted your idea, as they omit a lot of dependencies that are common to other packages of the same type (and from the same source), but we did not create a `buildGnomeApplication` for the Gnome package (although this may omit a lot of lines, sometimes it may also complicate things)

To be honest, I don’t think one is better than the other. Propagation has its own problems, which is why sometimes we want to get rid of it. This depends more on the maintainer's preference and the specific situation of the problem being solved, while this doesn't seem like a place where this solution is needed.